src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null) {
+            return false;
+        }
+        if (!o.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return (width == null && rectangle.width == null
```

Comment:
just a note: this may be simplified
```suggestion
        return (width == rectangle.width
```